modules/math/src/test/java/com/opengamma/strata/math/impl/interpolation/SmithWilsonCurveFunctionTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.math.impl.interpolation;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.util.function.Function;
+
+import org.testng.annotations.Test;
+
+import com.opengamma.strata.collect.DoubleArrayMath;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.math.impl.differentiation.FiniteDifferenceType;
+import com.opengamma.strata.math.impl.differentiation.ScalarFieldFirstOrderDifferentiator;
+import com.opengamma.strata.math.impl.differentiation.ScalarFirstOrderDifferentiator;
+
+/**
+ * Test {@link SmithWilsonCurveFunction}.
+ */
+@Test
+public class SmithWilsonCurveFunctionTest {
+
+  private static final double TOL = 1.0e-13;
+  private static final double ONE_BP = 1.0e-4;
+  private static final double EPS = 1.0e-7;
+  private static final ScalarFirstOrderDifferentiator DERIVATIVE =
+      new ScalarFirstOrderDifferentiator(FiniteDifferenceType.CENTRAL, EPS);
+  private static final ScalarFieldFirstOrderDifferentiator PARAM_SENSI =
+      new ScalarFieldFirstOrderDifferentiator(FiniteDifferenceType.CENTRAL, EPS);
+  private static final SmithWilsonCurveFunction SW_FUNCTION = SmithWilsonCurveFunction.DEFAULT;
+
+  public void test_function() {
+    DoubleArray weights = DoubleArray.of(2.66199573146653, -2.69394271301803, 0.484066828515597, 0.690098457608931,
+        -0.380409008626438, 0.228249649568978, -0.160902407241201);
+    DoubleArray nodes = DoubleArray.of(3d, 5d, 7d, 10d, 15d, 20d, 30d);
+    double alpha = 0.162009;
+    double[] dfExp = new double[] {
```

Comment:
Added comment "Source: EIOPA - European Insurance and Occupational Pensions Authority".